virtcontainers/netns_watcher/netns_watcher.c

```diff
@@ -0,0 +1,1251 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#define _GNU_SOURCE
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/wait.h>
+#include <fcntl.h>
+#include <sched.h>
+#include <asm/types.h>
+#include <sys/socket.h>
+#include <net/if.h>
+#include <netinet/in.h>
+#include <linux/netlink.h>
+#include <linux/rtnetlink.h>
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+#include <netdb.h>
+#include <sys/ioctl.h>
+#include <syslog.h>
+#include <getopt.h>
+
+#define PROGRAM_NAME "netns-watcher"
+
+#define MAX_IFACES  50
+#define INIT_IF_IDX -1
+
+bool debug = false;
+
+struct watcher_params {
+	char *pod_id;
+	char *runtime_path;
+	char *netns_path;
+};
+
+struct ip_addr {
+	unsigned char  family;
+	char           *addr;
+	struct ip_addr *next;
+};
+
+struct iface {
+	int            idx;
+	char           *hw_addr;
+	char           *name;
+	char           *mtu;
+	struct ip_addr *ip_addrs;
+};
+
+struct route {
+	char *src;
+	char *dst;
+	char *gw;
+	char *dev;
+};
+
+/* 
+ * Global interface list used to keep track of the interfaces and their
+ * parameters.
+ */
+struct iface iface_list[MAX_IFACES];
+
+/*
+ * Print full list of interfaces.
+ */
+void print_iface_list() {
+	int i;
+
+	for (i = 0; i < MAX_IFACES; i++) {
+		if (iface_list[i].idx == INIT_IF_IDX) {
+			continue;
+		}
+
+		printf("IFACE %d\n", i);
+		printf("\tidx     = %d\n", iface_list[i].idx);
+		printf("\thw_addr = %s\n", iface_list[i].hw_addr);
+		printf("\tname    = %s\n", iface_list[i].name);
+		printf("\tmtu     = %s\n", iface_list[i].mtu);
+
+		struct ip_addr *ipa = iface_list[i].ip_addrs;
+		if (!ipa) {
+			printf("\n");
+			continue;
+		}
+
+		printf("\tip_addrs\n");
+		printf("\t    |---- ip_addr = %s, family = %d\n",
+				ipa->addr, ipa->family);
+		while (ipa->next) {
+			printf("\t    |---- ip_addr = %s, family = %d\n",
+					ipa->next->addr, ipa->next->family);
+			ipa = ipa->next;
+		} 
+
+		printf("\n");
+	}
+}
+
+/*
+ * Print program usage.
+ */
+void print_usage(void)
+{
+	printf("Usage: %s [options]\n\n", PROGRAM_NAME);
+	printf(" -d, --debug        Enable debug output\n");
+	printf(" -h, --help         Display usage\n");
+	printf(" -n, --netns-path   Network namespace path (required)\n");
+	printf(" -p, --pod-id       Pod ID (required)\n");
+	printf(" -r, --runtime-path Runtime path (required)\n");
+	printf(" -v, --version      Show version\n");
+	printf("\n");
+}
+
+/*
+ * Print version information.
+ */
+void print_version(void)
+{
+	printf("%s v0.1\n", PROGRAM_NAME);
```

Comment:
- Hard-coded version number. How about a `#define` for it?
- Version should be in semver format.

Reply:
done